src/main/scala/viper/gobra/frontend/info/implementation/property/Implements.scala

```diff
@@ -104,7 +110,7 @@ trait Implements { this: TypeInfoImpl =>
         case Type.NilType | Type.BooleanT | _: Type.IntT | Type.StringT => true
         case ut: Type.PointerT => go(ut.elem)
         case ut: Type.StructT =>
-          ut.clauses.forall{ case (_, (_, fieldType)) => go(fieldType) }
+          ut.clauses.forall{ case (_, info) => go(info.typ) }
```

Comment:
structs with ghost fields are not identity preserving. This also makes line 100 redundant.